plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -247,49 +248,81 @@ public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStati
         }
     }
 
+    // Processing a single Partition, parallelize over pages
     @Override
-    public void updatePartitionStatistics(Partition partition, Map<String, HiveColumnStatistics> updatedColumnStatistics)
+    public void updatePartitionStatistics(Partition partition, Map<String, HiveColumnStatistics> columnStatistics)
```

Comment:
Is there a reason to not drop this method and just have `public void updatePartitionStatistics(Set<StatisticsUpdate> statisticsUpdates)`?

Reply:
The parallelism is a little different but I'm not sure it it matters. This is better if the table is very wide since it writes multiple pages of stats at a time, if the table is big enough to warrant it. So for a very wide, unpartitioned table it might be faster. Wdyt?

Reply:
Oh I see that now. It seems we can have column-chunk/column based paralelism here too. 
```java
    @Override
    public void updatePartitionStatistics(Set<StatisticsUpdate> statisticsUpdates)
    {
        List<CompletableFuture<Void>> operationsFutures = new ArrayList<>();
        for (StatisticsUpdate update : statisticsUpdates) {
            Partition partition = update.getPartition();
            Map<String, HiveColumnStatistics> updatedColumnStatistics = update.getColumnStatistics();

            HiveBasicStatistics partitionStats = getHiveBasicStatistics(partition.getParameters());
            List<ColumnStatistics> columnStats = toGlueColumnStatistics(partition, updatedColumnStatistics, partitionStats.getRowCount()).stream()
                    .filter(this::isGlueWritable)
                    .collect(toUnmodifiableList());

            List<List<ColumnStatistics>> columnChunks = Lists.partition(columnStats, GLUE_COLUMN_WRITE_STAT_PAGE_SIZE);

            columnChunks.forEach(columnChunk ->
                    operationsFutures.add(runAsync(() -> glueClient.updateColumnStatisticsForPartition(
                            new UpdateColumnStatisticsForPartitionRequest()
                                    .withCatalogId(catalogId)
                                    .withDatabaseName(partition.getDatabaseName())
                                    .withTableName(partition.getTableName())
                                    .withPartitionValues(partition.getValues())
                                    .withColumnStatisticsList(columnChunk)),
                            writeExecutor)));

            Map<String, HiveColumnStatistics> currentColumnStatistics = this.getPartitionColumnStatisticsIfPresent(partition).orElse(ImmutableMap.of());
            Set<String> removedStatistics = difference(currentColumnStatistics.keySet(), updatedColumnStatistics.keySet());
            removedStatistics.forEach(column ->
                    operationsFutures.add(runAsync(() -> glueClient.deleteColumnStatisticsForPartition(
                            new DeleteColumnStatisticsForPartitionRequest()
                                    .withCatalogId(catalogId)
                                    .withDatabaseName(partition.getDatabaseName())
                                    .withTableName(partition.getTableName())
                                    .withPartitionValues(partition.getValues())
                                    .withColumnName(column)),
                            writeExecutor)));
        }
        getFutureValue(allOf(operationsFutures.toArray(CompletableFuture[]::new)));
    }
```

Am I missing sth?